### DISCUSSION TITLE
Api does not maintain saperate sessions and uses same context and chat history for different sessions

### DISCUSSION THREAD

**Discussion by draco121** [safevideo/autollm#185] [Q&A] [answered]
I tried using the fastapi server and found out that when I call api from two different hosts , the query engine considers call coming from a single source and uses the existing context to answer the prompt, 
Ideally the query engine should be aware of different clients and maintain different contexts per client

**Comment by talhasarit** [comment, marked as answer]
Hello @draco121,

Thank you for reporting this issue. We have implemented enhancements in the latest version of AutoLLM that address your concerns:

- Unique session handling for different clients, assuring isolated contexts.
- Improved LanceDB URI validation and default database creation for new sessions.
- New parameters `exist_ok` and `overwrite_existing` for explicit database write protection.

These updates are part of our ongoing efforts to improve the robustness and usability of **autollm**.

Please update to the latest version to access these new features:

```bash
pip install --upgrade autollm
```

We appreciate your feedback and look forward to any further insights you may have.

Best,
Talha
